Moving the build logic into Dockerfiles + cleaning the code

liranmauda opened the pull request:
### Moving the build logic into Dockerfiles

Agent.Dockerfile
- Will build the agent without the need to supply noobaa-setup
- Adding the Git Commit hash to the package.json

builder.Dockerfile:
- Creating the noobaa/builder image that contains the dependencies for the build.

Tests.Dockerfile:
- Moved the dependencies building into builder.Dockerfile
- Moved the copy of the code infant of the build of the frontend due to dependencies

Server.Dockerfile:
- Building the server inline
- Adding the Git Commit hash to the package.json

Makefile:
- Created Makefile that builds builder, server, agent and test images
- Gets the git commit hash and passing it into the builder command

Added fix_package_json.sh for fixing the package.json if needed

### Renamed install_noobaa.sh to set_mongo_repo.sh and Cleaned the code

### Fixed dependencies order in frontend/gulpfile.js + Package.json changes

 frontend/gulpfile.js:
- Added  'install-deps' as dependencies for 'compile-styles'

package.json:
- build:fe is not calling npm_install.js anymore because
npm install is not calling rpm run build anymore

frontend/package.json:
- Removed the condition from the prepare

### Cleaning deploy base and creating setup_platform + Cleaning the code os_utils.js:

- Removed get_yum_proxy and set_yum_proxy

system_server.js:
- Removed the call for os_utils.set_yum_proxy

server_monitor.js:
- Cleaned the code from non-docker specific code.

noobaa_init.sh:
- Removed the extract of an rpm

- Removed deploy_base.sh
- Removed add_mongo_ssl_user.js
- Creating setup_platform.sh
- Removed setup_server_swap.sh

- Fixes for removing deploy_base.sh
- .dockerignore changes
- Removed npm_install.js

liranmauda committed:
Fixed dependencies order in frontend/gulpfile.js + Package.json changes

frontend/gulpfile.js:
- Added  'install-deps' as dependencies for 'compile-styles'

package.json:
- build:fe is not calling npm_install.js anymore because
npm install is not calling rpm run build anymore

frontend/package.json:
- Removed the condition from the prepare

liranmauda committed:
Renamed install_noobaa.sh to set_mongo_repo.sh and Cleaned the code

Renamed install_noobaa.sh to set_mongo_repo.sh and Cleaned the code

nimrod-becker commented on src/server/bg_services/server_monitor.js:
```diff
@@ -75,16 +73,8 @@ async function run_monitors() {
 
-    if (PLATFORM !== 'docker') {
-        await _verify_ntp_cluster_config();
-        await _verify_proxy_cluster_config();
```
Good change, but what will we see in the FE on the server page now we don't verify these statuses? We don't have to change the FE now, but may be set to always green until we do remove it?

liranmauda replied:
currently on the UI, it is shown as Not configured: 
![NooBaa_Management_Console](https://user-images.githubusercontent.com/28217223/59031276-e2d50400-886b-11e9-88ca-97eacf8753d3.png)

dannyzaken commented on src/deploy/NVA_build/Tests.Dockerfile:
```diff
@@ -63,27 +21,34 @@ RUN echo 'PATH=$PATH:/noobaa-core/node_modules/.bin' >> ~/.bashrc
+ARG GIT_COMMIT 
+RUN if [ "${GIT_COMMIT}" != "" ]; then sed -i 's/^  "version": "\(.*\)",$/  "version": "\1-'${GIT_COMMIT:0:7}'",/' package.json; fi
 
 FROM base as unitest
```
not necessarily unit tests. can be for test_env_builder too, isn't it?

dannyzaken commented on src/deploy/NVA_build/Tests.Dockerfile:
```diff
@@ -94,9 +59,8 @@ FROM base as unitest
-RUN yum install -y -q centos-release-scl && \
-    yum install -y -q rh-mongodb36 && \
-    yum install -y ntpdate && \ 
+RUN yum install -y -q rh-mongodb36 && \
```
`kubectl` is required to run test_env_builder

liranmauda committed:
Cleaning deploy base and creating setup_platform + Cleaning the code

os_utils.js:
- Removed get_yum_proxy and set_yum_proxy

system_server.js:
- Removed the call for os_utils.set_yum_proxy

server_monitor.js:
- Cleaned the code from non-docker specific code.

noobaa_init.sh:
- Removed the extract of an rpm

- Removed deploy_base.sh
- Removed add_mongo_ssl_user.js
- Creating setup_platform.sh
- Removed setup_server_swap.sh

- Fixes for removing deploy_base.sh
- .dockerignore changes
- Removed npm_install.js
- Removed run-agent.sh

liranmauda committed:
Moving the build logic into Dockerfiles

Agent.Dockerfile
- Will build the agent without the need to supply noobaa-setup
- Adding the Git Commit hash to the package.json

builder.Dockerfile:
- Creating the noobaa/builder image that contains the dependencies for
the build.

Tests.Dockerfile:
- Moved the dependencies building into builder.Dockerfile
- Moved the copy of the code infant of the build of the frontend due to
dependencies

Server.Dockerfile:
- Building the server inline
- Adding the Git Commit hash to the package.json

Makefile:
- Created Makefile that builds builder, server, agent and test images
- Gets the git commit hash and passing it into the builder command
